test/RelayServer.test.ts

```diff
@@ -1242,6 +1260,51 @@ contract('RelayServer', function (accounts) {
     // })
   })
 
+  describe('Alerted state as griefing mitigation', function () {
+    let newServer: RelayServer
+    beforeEach('should enter an alerted state for a configured blocks delay after paymaster rejecting an on-chain tx', async function () {
+      id = (await snapshot()).result
+      newServer = await bringUpNewRelay({ alertedBlockDelay: 100 })
+      await newServer._worker(await _web3.eth.getBlock('latest'))
+      await attackTheServer(newServer)
+    })
+    afterEach(async function () {
+      await revert(id)
+    })
+
+    async function attackTheServer (server: RelayServer): Promise<void> {
+      const _sendTransactionOrig = server._sendTransaction
+      const _sendTransaction = async function ({ signer, method, destination, value = '0x', gasLimit, gasPrice }: SendTransactionDetails): Promise<SignedTransactionDetails> {
+        await rejectingPaymaster.setRevertPreRelayCall(true)
+        // @ts-ignore
+        return (await _sendTransactionOrig.call(server, ...arguments))
+      }
+      server._sendTransaction = _sendTransaction
+      await relayTransaction(server, options, { paymaster: rejectingPaymaster.address }, false)
+      const currentBlock = await _web3.eth.getBlock('latest')
+      await server._worker(currentBlock)
+      assert.isTrue(server.alerted, 'server not alerted')
+      assert.equal(server.alertedBlock, currentBlock.number, 'server alerted block incorrect')
+    }
+
+    it('delay txs in alerted state', async function () {
+      newServer.minAlertedDelayMS = 300
+      newServer.maxAlertedDelayMS = 350
+      const timeBefore = Date.now()
+      await relayTransaction(newServer, options)
+      const timeAfter = Date.now()
+      assert.isTrue((timeAfter - timeBefore) > 300, 'checking that enough time passed')
+    })
+    it('should exit alerted state after the configured blocks delay', async function () {
+      await evmMineMany(newServer.alertedBlockDelay - 1)
+      await newServer._worker(await _web3.eth.getBlock('latest'))
+      assert.isTrue(newServer.alerted, 'server not alerted')
+      await evmMineMany(2)
+      await newServer._worker(await _web3.eth.getBlock('latest'))
+      assert.isFalse(newServer.alerted, 'server alerted')
+    })
+  })
+
   describe.skip('runServer', function () {
```

Comment:
I know this test was skipped before, just checking it is intentional

Reply:
Yes since it's a placeholder, will get to it this week.